hooks/honeybadger/honeybadger.go

```diff
@@ -0,0 +1,36 @@
+package logrus_honeybadger
+
+import (
+	"github.com/DavidHuie/gobadger"
+	"github.com/Sirupsen/logrus"
+)
+
+// HoneybadgerHook to send exceptions to an exception-tracking service compatible
+// with the Honeybadger API.
+type HoneybadgerHook struct {
+	conn *gobadger.Conn
+}
+
+func NewHoneyBadgerHook(apiKey string) *HoneybadgerHook {
+	return &HoneybadgerHook{gobadger.NewConn(apiKey)}
+}
+
+func (hook *HoneybadgerHook) Fire(entry *logrus.Entry) error {
+	err := hook.conn.Error(entry.Message)
```

Comment:
@agonzalezro : `context.Context` can't cancel a goroutine anymore than what you can do yourself from within the goroutine. To truly cancel the goroutine, you'd need something like Java's `Thread.interrupt()` or similar. This is not possible in Go by design. When people design their Go packages, they should keep that in mind and add a mean to configure timeouts or abort operations with `context.Context`.

The HoneyBadger client that is imported here is very simplistic and doesn't do that. I'd recommend using (or implementing, looks pretty simple) another one, one that can provide explicit timeouts. The http client used by this package is unexported and there's no way to configure it with custom timeouts.

https://github.com/DavidHuie/gobadger/blob/master/gobadger.go#L158

By default, the `http.Client` has a timeout of `0` which means it never times out.